server/mem_agent/engine.py

```
@@ -0,0 +1,334 @@
+            "Sandboxed code exceeded time limit of %d seconds; terminating.", timeout
+        )
+        return None, f"TimeoutError: Code execution exceeded {timeout} seconds."
+
+    if result.returncode != 0:
+        return None, result.stderr.decode().strip()
+
+    print("stderr:", result.stderr.decode())
+    print("stdout:", result.stdout[:200])
+    
+    try:
+        local_vars, error_msg = pickle.loads(result.stdout)
```

Ruff (0.14.1)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)